WIP: added new function to get one_time_correlation data from two_tim…

sameera2004 opened the pull request:
…e_corrrelation

ericdill commented:
This now has merge conflicts. It also needs a test

sameera2004 commented:
Sure, I will fix it and add a test to that

sameera2004 commented:
This PR is ready for review, added tests

ericdill commented on skbeam/core/correlation.py:
```diff
@@ -769,3 +769,29 @@ def _validate_and_transform_inputs(num_bufs, num_levels, labels):
+    ----------
+    two_time_corr : array
+        matrix of two time correlation
+        shape (number of images, number of images, number of labels(ROI))
```
I think your shape is backwards. Shouldn't this be `(num ROIs, num images, num images)`?  Same below in the `Returns` section

ericdill replied:
After in-person discussion with @sameera2004 , we have concluded that the two-time correlation results, that is `g2`, should have a shape of `(num ROI, num images, num images)` which will require touching the `_init_state_two_time()` function, and the `two_time_state_to_results()` function and probably the tests and the examples.... (sorry :cry: )

sameera2004 commented:
@ericdill I changed the `two-time correlation results`, that is `g2`, to have a shape of `(num ROI, num images, num images)` and fixed the tests too. They are passing and this PR is ready for review.

ericdill commented on skbeam/core/tests/test_correlation.py:
```diff
@@ -44,9 +44,12 @@
                                      lazy_one_time,
                                      lazy_two_time, two_time_corr,
-                                     two_time_state_to_results)
+                                     two_time_state_to_results,
```
This is imported twice

ericdill commented:
Did you consider having the one time from two time function return the same thing as `multi_tau_auto_corr`?

ericdill commented:
Err have it return the correlation and the lag steps?

ericdill commented:
Oh wait you already have that info from the two time results

sameera2004 commented:
@ericdill No `one_time_from_two_time_function` will not return the same results as `multi_tau_auto_corr`. If you `brute force`(not `multi tau`) using `multi tau auto corr` the results will have same shape.

sameera2004 commented:
@ericdill I fixed the PR ready to review

ericdill commented on skbeam/core/tests/test_correlation.py:
```diff
@@ -198,6 +200,28 @@ def test_bad_images():
+    roi[0:x_dim//10, 0:y_dim//10] = 5
+    roi[x_dim//10:x_dim//5, y_dim//10:y_dim//5] = 3
+
+    two_time = two_time_corr(roi, imgs, stack,
```
It is a little difficult to understand exactly what `two_time[0]` means in this test, as you need to understand trace the results of `two_time_corr` through about 3 function calls to understand what exactly it returns.  Having just realized that it is difficult to figure out what `two_time[0]` is, I just put in #407 to update the docs for `two_time_corr`.  Could you also change `two_time` to be `g2, lag_steps, _state` to make the meaning of the return value of `two_time_corr` a little clearer in this test?

ericdill commented:
Hi, one last minor comment. Then it is good to go!

sameera2004 commented:
@ericdill i fixed this PR ready for review. For your PR #407 I left a minor comment, other than it is good to go.

ericdill commented:
Awesome, thanks @sameera2004 !

sameera2004 commented:
@ericdill Thank you very much
